hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/TransferExecutor.java

```diff
@@ -130,17 +144,65 @@ protected void executeCryptoTransfer(
             CryptoTransferStreamBuilder recordBuilder,
             boolean skipCustomFee) {
         final var topLevelPayer = context.payer();
-        // Use the op with replaced aliases in further steps
         transferContext.validateHbarAllowances();
 
         // Replace all aliases in the transaction body with its account ids
         final var replacedOp = ensureAndReplaceAliasesInOp(txn, transferContext, context, validator);
+
         // Use the op with replaced aliases in further steps
-        final var steps = decomposeIntoSteps(replacedOp, topLevelPayer, transferContext, skipCustomFee);
+        final List<TransferStep> steps = new ArrayList<>();
+        steps.add(new AssociateTokenRecipientsStep(replacedOp));
+        final var customFeeStep = new CustomFeeAssessmentStep(replacedOp);
```

Comment:
This `CustomFeeAssessmentStep` doesn't seem to function as much like a `Step` as it does a singleton service that takes a `CryptoTransferTransactionBody` and its `TransferContext` and applies some business logic to those inputs. 🤔

Reply:
I have named all the things that happen in transfer execution as steps. 🤔